### DISCUSSION TITLE
How to Compile Changes to Code on MacOS?

### DISCUSSION THREAD

**Discussion by janvi-kalra** [flameshot-org/flameshot#3149] [Q&A]
Noob question, how do you test code changes made on a macOS?

I am able to run Flameshot successfully. Once I edit code in `circletool.cpp` in vscode, however, the changes don't show up. I tried to delete the `build` directory and re-build and re-install the code via 
```
mkdir build
cd build
cmake ../ -DQt5_DIR=$(brew --prefix qt5)/lib/cmake/Qt5
make
```
and then 
```
sudo make install
```
but no luck. 

For example, here is how I changed the code in `circletool.cpp`, but the tool still looks the same (the description, icon name, name did not change)
<img width="578" alt="Screenshot 2023-04-05 at 7 42 00 PM" src="https://user-images.githubusercontent.com/119091286/230259241-e0ee3c21-1df0-4886-ad96-03695e389aab.png">

<img width="927" alt="Screenshot 2023-04-05 at 7 42 19 PM" src="https://user-images.githubusercontent.com/119091286/230259230-ebff17f4-37c6-426a-8ade-a31bfdd6c4ec.png">

**Comment by mmahmoudian** [comment]
I don't think your problem is with compiling. It seems your problem is with running the version you have compiled. If you run the binary of flameshot in your build folder, you will see your changes. 

As a side note, you can always check which binary you are running by `ls -al "$(which flameshot)"`
